### PR TITLE
Added ParticipantsDatabase tests

### PR DESCRIPTION
**Pull request by juanlofer-eprosima**
Signed-off-by: Juan López Fernández <juanlopez@eprosima.com>

### CODE REVIEW

File: test/unittest/participant/participants_database/ParticipantsDatabaseTest.cpp [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+    ParticipantId id("void_part");
+    std::shared_ptr<VoidParticipant> participant = std::make_shared<VoidParticipant>(id);
+    // Insert participant
+    participants_database->add_participant(participant->id(), participant);
+    // Reinsert to assure there is no error when inserting duplicates, only a warning is printed
+    participants_database->add_participant(participant->id(), participant);
+
+    auto participants = participants_database->get_participants_map();
+    auto it = participants.find(id);
+    ASSERT_EQ(it->second, participant);
```
[the remark this answers is not in the document]
Comment (rsanchez15) [reply test/unittest/participant/participants_database/ParticipantsDatabaseTest.cpp:100]: So I see three checks every time you add or remove a participant from the database:
- Check if the key (participant id) is in the database: `ASSERT_TRUE(participants_database->get_participants_map().count(id));`
- Check that the participant in that key is the same you saved: `ASSERT_EQ(it->second, participant);`
- Check the size of the map: `ASSERT_EQ(participants_database->size(), X);`

**Review by rsanchez15 — CHANGES_REQUESTED** [review]

File: test/unittest/participant/participants_database/ParticipantsDatabaseTest.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (juanlofer-eprosima) [reply test/unittest/participant/participants_database/ParticipantsDatabaseTest.cpp:100]: :+1:

[the remark this answers is not in the document]
Comment (juanlofer-eprosima) [reply test/unittest/participant/participants_database/ParticipantsDatabaseTest.cpp]: :+1:

File: test/unittest/participant/participants_database/ParticipantsDatabaseTest.cpp [hunk 2]
```diff
@@ -0,0 +1,220 @@
...
+    std::shared_ptr<VoidParticipant> participant = std::make_shared<VoidParticipant>(id);
+    // Pop not yet stored participant
+    ASSERT_TRUE(participants_database->pop(id) == nullptr);
+
+    // Insert participant into database
+    participants_database->add_participant(participant->id(), participant, 1);
+    ASSERT_EQ(participants_database->get_participant(id), participant);
+
+    // Pop stored participant
+    ASSERT_EQ(participants_database->pop(id), participant);
```
Comment (rsanchez15) [inline test/unittest/participant/participants_database/ParticipantsDatabaseTest.cpp:133]: Test the same but using the `pop()` function.

**Review by rsanchez15 — CHANGES_REQUESTED** [review]

File: test/unittest/participant/participants_database/ParticipantsDatabaseTest.cpp (hunk 2 above)
Comment (juanlofer-eprosima) [reply test/unittest/participant/participants_database/ParticipantsDatabaseTest.cpp]: :+1:

**Review by rsanchez15 — APPROVED** [review]

**Pull request merged**